File: front-end/src/schemas/message.ts
import * as v from 'valibot';

export const messageSchema = v.object({
  id: v.nullable(
    v.number(),
  ),
  name: v.pipe(
    v.string(),
    v.minLength(1, 'Name is required'),
  ),
  enabled: v.boolean(),
  channelId: v.pipe(
    v.string(),
    v.trim(),
    v.minLength(1, 'Channel is required'),
  ),
  channelName: v.nullish(
    v.string(),
  ),
  messageTemplate: v.pipe(
    v.string(),
    v.trim(),
    v.minLength(1, 'Message template is required'),
  ),
  shouldSendTemplate: v.pipe(
    v.string(),
    v.trim(),
    v.minLength(1, 'Should Send template is required'),
  ),
  timeoutTemplate: v.pipe(
    v.string(),
    v.trim(),
    v.minLength(1, 'Timeout template is required'),
  ),
  stationId: v.nullable(
    v.pipe(
      v.number(),
      v.minValue(0, 'Station is required'),
      v.transform((x) => x === 0 ? null : x),
    ),
  ),
  botId: v.pipe(
    v.number('Bot is required'),
  ),
});

export type MessageType = v.InferInput<typeof messageSchema>;

File: python-shared/shared/models/odmantic/message.py
from odmantic import Model, Reference, Field
from datetime import datetime
from typing import Optional, List

from .bot import Bot
from .station import Station


class Message(Model):
    channel_id: Optional[str] = None
    name: Optional[str] = None
    message_template: Optional[str] = None
    should_send_template: Optional[str] = None
    timeout_template: Optional[str] = None

    bot: Optional[Bot] = Reference()
    last_sent_time: Optional[datetime] = None
    enabled: Optional[bool] = True

    stations: List[Station] = Field(default_factory=list)

    class Config:
        collection = "message"

    def __str__(self):
        station_ids = [str(s.id) for s in self.stations] if self.stations else []
        return (
            f"Message(id={self.id}, channel_id='{self.channel_id}', "
            f"name='{self.name}', stations={station_ids}, "
            f"last_sent_time={self.last_sent_time}, enabled={self.enabled})"
        )
